### PR TITLE
Add `next_gen_ui_demo` to samples index

### PR DESCRIPTION
**Pull request by domesticmouse**
## Pre-launch Checklist

- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Contributors Guide]: https://github.com/flutter/samples/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by craiglabenz — APPROVED** [review]
LGTM

**Comment by domesticmouse** [comment]
@johnpryan I'm failing to find the published demo. I'm guessing I broke something.

File: web/samples_index/lib/src/samples.yaml [hunk 1]
```diff
@@ -653,3 +653,19 @@ samples:
...
+        alt: Next Gen UI demo screenshot
+    source: https://github.com/flutter/samples/tree/main/next_gen_ui_demo
+    description: A sample version of the Next Generation UI codelab for demoing at I/O Connect
+    difficulty: advanced
+    packages: 
+      - flutter_animate
+    platforms:
+      - Windows, macOS, Linux, web, Android, iOS
+    links:
+      - https://codelabs.developers.google.com/codelabs/flutter-next-gen-uis
```
Comment (johnpryan) [inline web/samples_index/lib/src/samples.yaml:669]: Links are a map with two entries,`text` and `href`. https://github.com/flutter/samples/blob/main/web/samples_index/lib/src/data.dart#L178

The [GitHub Action](https://github.com/flutter/samples/actions/runs/5011332922/jobs/8982046277) shows the exception:

```
ParsedYamlException: line 669, column 7 of lib/src/samples.yaml: Unsupported value for "links". type 'String' is not a subtype of type 'Map<dynamic, dynamic>' in type cast
    ╷
669 │ ┌       - https://codelabs.developers.google.com/codelabs/flutter-next-gen-uis
670 │ │     type: sample
    │ └────^
    ╵
#0      checkedYamlDecode (package:checked_yaml/checked_yaml.dart:48:5)
#1      getSamples (package:samples_index/samples_index.dart:18:15)
<asynchronous suspension>
#2      generate (file:///home/runner/work/samples/samples/web/samples_index/tool/grind.dart:46:17)
<asynchronous suspension>
#3      _RootZone.bindUnaryCallbackGuarded.<anon> (dart:async/zone.dart:1633:12)
<asynchronous suspension>
  Failed! (1) – samples_index – `flutter pub run grinder deploy`
```

Comment (domesticmouse) [reply web/samples_index/lib/src/samples.yaml:669]: Looking at the content of the config file it looks like zero entries currently use the `links` attribute. Is this a capability that is unsupported, or should be sunset due to lack of use?

Also, debugging something that happens after commit is less than optimal.

**Pull request merged**